### PR TITLE
Ndelangen/temp-alternative-solution-react-dom

### PR DESCRIPTION
**Pull request by ndelangen**
@shilman I think this fixes the react-dom v18 issues we looked at with @valentinpalkovic

### CODE REVIEW

**Comment by ndelangen** [comment]
we could move this code into @valentinpalkovic's branch too

File: app/react/src/client/preview/render.tsx [hunk 1]
```diff
@@ -26,10 +31,56 @@ export const render: ArgsStoryFn<ReactFramework> = (args, context) => {
...
+
+const getReactRoot = async (el: Element): Promise<ReactRoot | null> => {
+  if (!isUsingNewReactRootApi) {
+    return null;
+  }
+
+  let root = nodes.get(el);
+
+  if (!root) {
+    // Skipping webpack's static analysis of import paths by defining the path value outside the import statement.
```
Comment (valentinpalkovic) [inline app/react/src/client/preview/render.tsx:74]: We can remove this comment

**Comment by ndelangen** [comment]
Looks like there still some issues, @valentinpalkovic

File: app/react/src/server/framework-preset-react-dom-hack.ts [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+    ...config,
+    plugins: [
+      ...config.plugins,
+      reactDomPkg.version.startsWith('18')
+        ? null
+        : new IgnorePlugin({
+            resourceRegExp: /react-dom\/client$/,
+            contextRegExp: /(app\/react|@storybook\/react)/, // TODO this needs to work for both in our MONOREPO and in the user's NODE_MODULES
+          }),
+    ],
```
Comment (valentinpalkovic) [inline app/react/src/server/framework-preset-react-dom-hack.ts:20]: `].filter(Boolean)` is missing here

Comment (ndelangen) [reply app/react/src/server/framework-preset-react-dom-hack.ts:20]: Ah, yes. would you mind adding that? You should have access to push to this branch / or if you prefer merge this work unto your branch, continue there, and close this PR?

Comment (valentinpalkovic) [reply app/react/src/server/framework-preset-react-dom-hack.ts:20]: Sure!

**Comment by valentinpalkovic** [comment]
Closing PR. Continuing on https://github.com/storybookjs/storybook/pull/17215

**Pull request closed without merging**